genie-web/src/main/java/com/netflix/genie/web/services/impl/JobResolverServiceImpl.java

```diff
@@ -466,15 +472,16 @@ private void resolveCluster(final JobResolutionContext context) throws GenieJobR
                         );
                     }
                 } catch (final Exception e) {
-                    selectorTags.add(NO_CLUSTER_RESOLVED_ID);
-                    selectorTags.add(NO_CLUSTER_RESOLVED_NAME);
+                    // Selector runtime exception
+                    // To avoid non-determinism, don't proceed to the next one
```

Comment:
I'm not sure if this is our best option. We're basically choosing aid in debugging over prioritizing getting the users job to run in any manner possible. I kind of feel like it should be the other way around if a user has given a valid request and we have another option available to us

Reply:
The case here is:
1. Script selector throws runtime exception
2. Fallback to random selector
3. Random resource gets selected 

It's certainly a complex tradeoff with no clear winner.
Non-determinism in general seems pretty bad, so I opted for it here, in the specific case of runtime errors.

Related: why does this resolver iterates over selectors, while the other one does not?
Seems to me like they should both do it or neither. What am i missing?

Reply:
Originally the command one was intended to do the same fallback mechanism however our users stated that they didn't want this behavior. If I recall it was because they felt while cluster selection could afford to be random as generally they're selecting something that just provides resources and the users job will run one way or another in the case of commands changing the actual binaries that a job runs with could have unintended consequences (e.g user gets a buggy version or a version their code isn't compatible with etc). They didn't want to take that risk and this allowed them to use the script to blacklist versions or whatnot. I believe the original design doc has some of this discussion in the notes though I haven't checked.

Reply:
Thanks for the clarification.

Would you like me to reinstate the logic to "in case of script runtime error, proceed to the next selector"?

Reply:
yeah lets for now

Reply:
Done